experiment/simulation/js/main.js

```diff
@@ -1 +1,258 @@
-//Your JavaScript goes in here
+let detergentAddCounter = 1;
+
+// disable onclick of all components except of spoon and detergent
+if(document.getElementsByClassName("clothWithStain")[0]){
+    document.getElementsByClassName("clothWithStain")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("glassRod")[0]){
+    document.getElementsByClassName("glassRod")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("spoon2")[0]){
+    document.getElementsByClassName("spoon2")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("detergent2")[0]){
+    document.getElementsByClassName("detergent2")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("spoon3")[0]){
+    document.getElementsByClassName("spoon3")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("detergent3")[0]){
+    document.getElementsByClassName("detergent3")[0].style.pointerEvents = 'none';
+}
+
+
+// Functions for structure of soap part.
+// below function shows information about micelle parts when hovered upon
+function showMicelleInformation(part){
+    'use strict';
+    let tailInfo = document.getElementsByClassName("micelle-information")[0];
```

Comment:
this should be declared cosnt. cost is the default type, let is used only when using const gives error